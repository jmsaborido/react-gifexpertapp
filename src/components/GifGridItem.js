import React from 'react'

export const GifGridItem = ({ id, title, url }) => {
    return (
        <div>
            <img src={url} alt="No hay gif amigo" className="gif"></img>
            <p>{title}</p>
        </div>
    )
}
