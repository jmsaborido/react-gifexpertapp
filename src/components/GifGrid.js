import React, { useEffect, useState } from 'react'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const [images, setImages] = useState([])

    useEffect(() => {
        getGifs();
    }, [])

    const getGifs = async () => {
        const url = `https://api.giphy.com/v1/gifs/search?q=Rick+And+Morty&limit=10&api_key=${process.env.REACT_APP_API_KEY}`;
        const respuesta = await fetch(url);
        const { data } = await respuesta.json();
        const gifs = data.map(img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        })
        setImages(gifs);
    }

    return (
        <div>
            <h3>{category}</h3>
            <ol>{
                images.map((img) =>
                    <GifGridItem key={img.id} {...img} />
                )
            }</ol>
        </div>
    )
}
