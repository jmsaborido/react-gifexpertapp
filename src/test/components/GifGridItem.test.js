import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGridItem } from "../../components/GifGridItem";

describe('Pruebas en <GifGridItem />', () => {

    const title = 'Titulo';
    const url = 'https://localhost/test.jpg';

    test('debe de mostrar <GifGridItem /> correctamente', () => {
        const wrapper = shallow(<GifGridItem title={title} url={url} />)
        expect(wrapper).toMatchSnapshot();
    })
})