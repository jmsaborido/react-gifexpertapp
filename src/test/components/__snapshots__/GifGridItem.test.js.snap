// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe de mostrar <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="No hay gif amigo"
    className="gif"
    src="https://localhost/test.jpg"
  />
  <p>
    Titulo
  </p>
</div>
`;
